lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -1237,9 +1238,9 @@ def certificate_issued_for_csr(self, app_name: str, csr: str) -> bool:
             bool: True/False depending on whether a certificate has been issued for the given CSR.
         """
         issued_certificates_per_csr = self.get_issued_certificates()[app_name]
-        for request, cert in issued_certificates_per_csr.items():
-            if request == csr:
-                return csr_matches_certificate(csr, cert)
+        for request_pair in issued_certificates_per_csr:
```

Comment:
```suggestion
        for issued_pair in issued_certificates_per_csr:
```

Reply:
I'm not sure I get it, but any entry in `issued_certificates_per_csr` is a pair of certificate that was issued for the csr, isn't it?
What we do here, is we get a csr as a parameter, and we get all issued certificates, and we check if this given cert had a certificate issued for it before or not, if so we check it the issued certificate is correct (actually matches the csr).

Reply:
I get it now, I've changed it.